cli/integration/tests/waiter/test_cli.py

```diff
@@ -230,3 +232,24 @@ def update_token_loop():
             thread.join()
             self.logger.info('Thread finished')
             util.delete_token(self.waiter_url, token_name)
+
+    @unittest.skipIf('WAITER_TEST_CLI_COMMAND' in os.environ, 'waiter executable may be unknown.')
+    def test_base_config_file(self):
+        token_name = self.token_name()
+        cluster_name_1 = str(uuid.uuid4())
+        config = {'clusters': [{"name": cluster_name_1, "url": self.waiter_url}]}
+        with cli.temp_base_config_file(config):
```

Comment:
Will it be possible to write the config file to every location - e.g. what if the cli command is at `/usr/bin`?

Reply:
That's the reason for the `skipIf('WAITER_TEST_CLI_COMMAND' in os.environ, ...)` -- we only want to run this test when we're using the local (i.e. from `virtualenv` or `conda`) CLI. In cases where the tester is providing his own path, all bets are off for this test.

Reply:
Is that something we should check and have this test skipped?

Reply:
I'm not sure if I understand your question, but I think we should be pretty safe if the user doesn't specify `WAITER_TEST_CLI_COMMAND`, because we just use `waiter`, which will be locally installed any time these tests run.

Reply:
If I am testing against a production deployment and specify it using `WAITER_TEST_CLI_COMMAND`, the location will not be writable for the base config. Is that an issue we should worry about and avoid running this test?

Reply:
Yep, that's already happening with the `skipIf`:

`@unittest.skipIf('WAITER_TEST_CLI_COMMAND' in os.environ, 'waiter executable may be unknown.')`